### PR TITLE
[DF][Doc][NFC] Improve documentation of RNode and AsRNode

### PR DESCRIPTION
**Pull request by stwunsch**

### CODE REVIEW

File: tree/dataframe/src/RDataFrame.cxx [hunk 1]
```diff
@@ -481,6 +481,20 @@ sum = df.Filter("myFilter(x)").Sum("y")
 print(sum.GetValue())
 ~~~
 
+Alternatively, you can also pass the full RDataFrame object to C++ using the ROOT::RDF::AsRNode helper in Python, which casts any RDataFrame node to ROOT::RDF::RNode:
+
+~~~{.python}
+ROOT.gInterpreter.Declare('''
+ROOT::RDF::RNode MyTransformation(ROOT::RDF::RNode& df) {
```
Comment (eguiraud) [inline tree/dataframe/src/RDataFrame.cxx:488]: I am not sure taking by non-const reference would compile as we are passing in a temporary. But passing by value is totally ok (we also return by value so it's also good to be consistent I guess): 

```suggestion
ROOT::RDF::RNode MyTransformation(ROOT::RDF::RNode df) {
```

File: tree/dataframe/src/RDataFrame.cxx [hunk 2]
```diff
@@ -481,6 +481,20 @@ sum = df.Filter("myFilter(x)").Sum("y")
...
+    auto myFunc = [](float x){ return -x;};
+    return df.Define("y", myFunc, {"x"});
+}
+''')
```
Comment (eguiraud) [inline tree/dataframe/src/RDataFrame.cxx:492]: we use `"""` elsewhere I think

File: tree/dataframe/src/RDataFrame.cxx [hunk 3]
```diff
@@ -694,6 +708,16 @@ RDataFrame d("myTree", "file.root");
 std::vector<std::string> colNames = d.GetColumnNames();
 ~~~
 
+### Organizing RDataFrame in collections
+
+Any RDataFrame node can be converted to the common type ROOT::RDF::RNode. The cast can happen either implicitly by the C++ casting rules or explicitly using ROOT::RDF::AsRNode. This allows to organize RDataFrame nodes in any collection, e.g. a std::vector<ROOT::RDF::RNode>. Note that this adds an extra virtual call during the RDataFrame event loop any have an impact on the performance.
```
Comment (eguiraud) [inline tree/dataframe/src/RDataFrame.cxx:713]: In C++ there is no need for `AsRNode`, you can just construct an `RNode` from any RDF object:
```suggestion
Any RDataFrame node can be converted to the common type ROOT::RDF::RNode. The cast can happen either implicitly by the C++ casting rules or by passing any dataframe object to RNode's constructor. This allows to organize RDataFrame nodes in any collection, e.g. a `std::vector<ROOT::RDF::RNode>`. Note that this adds an extra virtual call during the RDataFrame event loop and can have a minor performance impact.
```

**Review by eguiraud — COMMENTED** [review]
Thanks @stwunsch ! Some minor comments below .

**Comment by stwunsch** [comment]
All fixed and squashed the changes in a single commit.

**Pull request merged**